building_sim_plugins/building_ignition_plugins/src/door.cpp

```diff
@@ -65,18 +65,21 @@ class IGNITION_GAZEBO_VISIBLE DoorPlugin
     // TODO proper rclcpp init (only once and pass args)
     auto model = Model(entity);
     char const** argv = NULL;
+    std::string name;
+    auto door_ele = sdf->GetElementImpl("door");
+    get_sdf_attribute_required<std::string>(door_ele, "name", name);
     if (!rclcpp::is_initialized())
       rclcpp::init(0, argv);
-    std::string plugin_name("plugin_" + model.Name(ecm));
+    std::string plugin_name("plugin_" + name);
```

Comment:
This is because, without nested model, the cabin door plugin is directly under the lift model. In such cases, `model.Name(ecm)` is going to return the name of the lift model instead of the cabin door. 
This should no longer be needed when nested models are supported in ignition.